rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java

```diff
@@ -626,6 +628,83 @@ public static ShallowClass build(String fullyQualifiedName) {
         }
     }
 
+    @Data
+    class AnnotationValue {
+        private final Method method;
+        private final String value;
+    }
+
+    class Annotation extends FullyQualified {
+
+        public final List<AnnotationValue> values;
+        public final FullyQualified type;
```

Comment:
I wonder if `@Delegate` might make sense here; did you explore that option?

Reply:
No, let me look into it.

Reply:
https://projectlombok.org/features/experimental/Delegate

> Current status: negative - Currently we feel this feature will not move out of experimental status anytime soon, and support for this feature may be dropped if future versions of javac or ecj make it difficult to continue to maintain the feature.

Sounds like it is a experimental lombok feature that might get deprecated in future. Probably should avoid using that?

Reply:
Now I understand why we implement `FullyQualified`: It is to remain compatible with the existing API. That is indeed a good reason.